### ISSUE TITLE
Deprecate support for Logback Layout

### ISSUE THREAD

**Issue by brenuart** [logstash/logstash-logback-encoder#638]
Logback has deprecated the use of `Layout` in favour of `Encoder`. A `LayoutWrappingEncoder` adapter is provided to bridge between the two.

Logback `OutputStreamAppender` and sub-classes like `ConsoleAppender` still accept a `Layout` but wrap them with a `LayoutWrappingEncoder` and emit warnings saying:

```
This appender no longer admits a layout as a sub-component, set an encoder instead.
To ensure compatibility, wrapping your layout in LayoutWrappingEncoder.
See also http://logback.qos.ch/codes.html#layoutInsteadOfEncoder for details.
```

The SyslogAppender however is still using a Layout but does not allow to set it by configuration.

Supporting Layout in logstash-logback-encoder is cumbersome and increase code complexity for little to no benefit. Proposition is to deprecate it for removal in the next major release.

**Comment by philsttr** [comment]
> The SyslogAppender however is still using a Layout but does not allow to set it by configuration.

`AbstractLogstashUdpSocketAppender` extends `SyslogAppenderBase`, and allows the layout to be configured.  It's typically configured with a Layout from logstash-logback-encoder.

If layout support is removed from logstash-logback-encoder, we'll need to copy the logic from `SyslogAppenderBase` into `AbstractLogstashUdpSocketAppender`

**Comment by brenuart** [comment]
> AbstractLogstashUdpSocketAppender extends SyslogAppenderBase...

Correct. We should definitely see how we can make the UdpSocketAppender use an Encoder (or else) before deprecating Layouts...
